src/libs/actions/Search.ts

```diff
@@ -83,4 +86,23 @@ function deleteMoneyRequestOnSearch(hash: number, transactionIDList: string[]) {
     API.write(WRITE_COMMANDS.DELETE_MONEY_REQUEST_ON_SEARCH, {hash, transactionIDList}, {optimisticData, finallyData});
 }
 
-export {search, createTransactionThread, deleteMoneyRequestOnSearch, holdMoneyRequestOnSearch, unholdMoneyRequestOnSearch};
+type Params = Record<string, string | string[]>;
+
+function exportSearchItemsToCSV(query: string, reportIDList: Array<string | undefined> | undefined, transactionIDList: string[], policyIDs: string[]) {
+    const fileName = `Expensify_${query}.csv`;
+
+    const finalParameters = enhanceParameters(READ_COMMANDS.EXPORT_SEARCH_ITEMS_TO_CSV, {
+        query,
+        reportIDList,
+        transactionIDList,
+        policyIDs,
+    }) as Params;
+
+    // Convert finalParameters to a query string
+    const queryString = Object.entries(finalParameters)
+        .map(([key, value]) => `${key}=${Array.isArray(value) ? value.join(',') : String(value)}`)
+        .join('&');
```

Comment:
Hmm couldn't this hit the url limit of 2048 chars? Each transactionID is ~19 chars long, so if the user selects ~100 transactions we'll hit that limit already, no?

Reply:
That is a good point, @filip-solecki I think we will have to go back to the previous idea of finding a way to send a post in the native apps for downloads.

Reply:
I had this issue at prev full time work. I remember writing a hack to create an `a` tag, setting the post response as the target for the `a` tag and clicking it using javascript 

```
const response = postRequest(url, contentType: 'text/csv', body);
const blob = await response.blob();

const a = document.createElement('a'); 
a.href = URL.createObjectURL(blob);
a.download = 'fileName.csv';

document.body.appendChild(a);
a.click(); 
```

Reply:
Are we sure that the lib can't handle post requests? It seems to define [other types](https://github.com/RonRadtke/react-native-blob-util/blob/master/index.d.ts#L628) and handle a [body param](https://github.com/RonRadtke/react-native-blob-util/blob/master/index.d.ts#L13) in fetch

Reply:
We are using the `downloadManager` which [according to the docs doesn't support a POST](https://github.com/RonRadtke/react-native-blob-util/tree/master#:~:text=also%20notice%20that%20Download%20Manager%20can%20only%20support%20GET%20method%2C%20which%20means%20the%20request%20body%20will%20be%20ignored.).

I think we should try to download without the `downloadManager` set to true and check what happens, but let's do it in the other issue so we can get this done by EOW.

Reply:
I think it's worth a shot with `useDownloadManager: false` the downside is that it won't show progress, notifications, etc. 

Is the issue only on android?

Reply:
cc @Julesssss maybe you might have some insights on using android's Download Manager and passing params to the request (not in the URL)

Reply:
both Android and iOS. Desktop also needs refactor but it is more likely to fix it quickly there

Reply:
I don't recall any specifics regarding the download manager GET limitation. Is it essential we use the native download manager lib? it has nice functionality but perhaps we can use an alternative method to download the CSV file (I think that is what we are trying to do here?) then show our own custom notification

Reply:
Exactly, this is what I am working on right now, downloading seems to be working on iOS right now